### PR TITLE
Adding config to add partner attribution

### PR DESCRIPTION
**Pull request by isha97**

### CODE REVIEW

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java [hunk 1]
```diff
@@ -115,6 +118,7 @@ public String getConnectorInfo() {
     jsonObject.addProperty("sparkVersion", SPARK_VERSION);
     jsonObject.addProperty("javaVersion", JAVA_VERSION);
     jsonObject.addProperty("scalaVersion", SCALA_VERSION);
+    partner.ifPresent(s -> jsonObject.addProperty("partner", s));
```
Comment (davidrabinowitz) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java:121]: replace partner with `GPN`

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
The change should be in `getUserAgent()` as well, as `(GPN:<partner>)` if such exists

File: spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/SparkBigQueryConfigTest.java [hunk 2]
```diff
@@ -232,6 +234,7 @@ public void testConfigFromOptions() {
     assertThat(config.getKmsKeyName()).isEqualTo(Optional.of("some/key/name"));
     assertThat(config.getAllowMapTypeConversion()).isFalse();
     assertThat(config.getBigQueryJobTimeoutInMinutes()).isEqualTo(30);
+    assertThat(config.getpartner().get()).isEqualTo("GPN:testUser");
```
[the remark this answers is not in the document]
Comment (isha97) [reply spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/SparkBigQueryConfigTest.java]: Fixed

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java (hunk 1 above)
Comment (isha97) [reply spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java]: Done

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**